spark/src/main/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteredTableUtils.scala

```diff
@@ -355,11 +355,29 @@ trait ClusteredTableUtilsBase extends DeltaLogging {
   private def validateClusteringColumnsInStatsSchema(
       statsCollection: StatisticsCollection,
       clusteringColumnInfos: Seq[ClusteringColumnInfo]): Unit = {
-    val missingColumn = getClusteringColumnsNotInStatsSchema(statsCollection, clusteringColumnInfos)
-    if (missingColumn.nonEmpty) {
+    val missingColumns =
+      getClusteringColumnsNotInStatsSchema(statsCollection, clusteringColumnInfos)
+    if (missingColumns.nonEmpty) {
+      // Check DataType eligibility.
+      val missingColumnSet = missingColumns.toSet
```

Comment:
nit: doesn't look like set is needed as Seq also supports contains. Also there'll be at most 4 clustering columns

Reply:
fixed